projects/gloo/pkg/api/converters/kube/account_credentials_secret.go

```
@@ -0,0 +1,75 @@
+package kubeconverters
+
+import (
+	"context"
+
+	"github.com/solo-io/solo-kit/pkg/utils/kubeutils"
+
+	v1 "github.com/solo-io/gloo/projects/gloo/pkg/api/v1"
+	"github.com/solo-io/solo-kit/pkg/api/v1/clients/kubesecret"
+	"github.com/solo-io/solo-kit/pkg/api/v1/resources"
+	skcore "github.com/solo-io/solo-kit/pkg/api/v1/resources/core"
+	kubev1 "k8s.io/api/core/v1"
+)
+
+const (
+	AccountCredentialsSecretType kubev1.SecretType = "extauth.solo.io/accountcredentials"
+	UsernameDataKey                                = "username"
+	PasswordDataKey                                = "password"
+)
+
+type AccountCredentialsSecretConverter struct{}
+
+func (t *AccountCredentialsSecretConverter) FromKubeSecret(_ context.Context, _ *kubesecret.ResourceClient, secret *kubev1.Secret) (resources.Resource, error) {
+	if secret.Type == AccountCredentialsSecretType {
+		username, ok := secret.Data[UsernameDataKey]
+		if !ok {
+			return nil, nil
```

Should we return an error here? Same on L31?

---

The pattern for these converters is that we don't return errors on mismatched types. We find the type of a secret by trying to convert it to each type that we know about and if it doesn't match one we try the next. (see https://github.com/solo-io/gloo/blob/9cd9e7b42a8dd1c9bc147feb4290363083303b04/projects/gloo/pkg/api/converters/kube/secretconverter.go)